### PR TITLE
add image tags 

### LINKED ISSUE #6: Image Tags
#### As a user I want to add tags to images.

__Story__:

Our users are loving the ability to add images to our service. However, now
that there are more than a handful of images, it is apparent that we need some
way to filter images by certain categories. In order to do that, we first need
to label images in some way. It is time to add the ability for our users to tag
images when they add them.

__Acceptance criteria__:
- [ ] When I add a new image, there is a form field to add tags.
- [ ] After I've saved an image with tags, I see those tags displayed with the
  associated image.
- [ ] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

__Reference material__:
- https://github.com/mbleigh/acts-as-taggable-on

__Note__: You should allow images to have no tags; this sets the stage for a later issue.

__Dependencies__:
- Image Index

### PR DESCRIPTION
**Pull request by stanleyphu**
Closes #6

---

The following is a checklist to remind you of the steps to take when opening a
new PR. The horizontal line above and everything below can be removed from the
pull request description.

- [ ] I have replaced `X` above with the associated issue number.
- [ ] I have changed the issue's label to "pull request".
- [ ] I have assigned the designated reviewer to this PR.
- [ ] I have set the appropriate label on this PR.
- [ ] I have written clear commit messages ([ref](http://chris.beams.io/posts/git-commit/)).

If needed, please review the document on the
[Image Sharing Project](https://sites.google.com/a/appfolio.com/eng/new-hire-resources/engineering-academy/ropes/image-sharing-project)
for information on how labels should be used.

### CODE REVIEW

File: test/controllers/images_controller_test.rb [hunk 1]
```diff
@@ -28,6 +28,19 @@ def test_show
...
+  def test_show_with_tags
+    image = Image.create(title: 'Canyon', url: 'https://www.picture.com', tag_list: 'tag1, tag2')
+
+    get image_path(image.id)
+
+    assert_response :success
+    assert_select '.js-title', image.title
+    assert_select 'img' do
+      assert_select '[src=?]', image.url
+    end
```
Comment (shiboying) [inline test/controllers/images_controller_test.rb]: This you can also do: `assert_select 'img[src=?]', image.url`

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -48,6 +61,21 @@ def test_create
...
+    assert_difference 'Image.count', 1 do
+      post images_url, params: { image: { title: 'test', url: 'https://www.picture.com', tag_list: 'tag_a' } }
+
+      assert_response :redirect
+      follow_redirect!
+      assert_response :success
+      assert_select '.js-title', 'test'
+      assert_select 'img' do
+        assert_select '[src=?]', 'https://www.picture.com'
+      end
```
Comment (shiboying) [inline test/controllers/images_controller_test.rb]: And here `assert_select 'img[src=?]', 'https://www.picture.com'`

File: test/controllers/images_controller_test.rb (hunk 1 above)
Comment (stanleyphu) [reply test/controllers/images_controller_test.rb]: fixed

File: test/controllers/images_controller_test.rb (hunk 2 above)
Comment (stanleyphu) [reply test/controllers/images_controller_test.rb]: fixed

**Pull request merged**